SerialPrograms/Source/PokemonSV/Programs/Farming/PokemonSV_MaterialFarmerTools.cpp

```diff
@@ -276,92 +369,16 @@ void run_one_sandwich_iteration(
-        }
-
-        // heal before starting Let's go
-        env.console.log("Heal before starting Let's go", COLOR_PURPLE);
-        env.console.log("Heal threshold: " + tostr_default(options.AUTO_HEAL_PERCENT), COLOR_PURPLE);
-        double hp = hp_watcher.last_known_value() * 100;
-        if (0 < hp){
-            env.console.log("Last Known HP: " + tostr_default(hp) + "%", COLOR_BLUE);
-        }else{
-            env.console.log("Last Known HP: ?", COLOR_RED);
-        }
-        if (0 < hp && hp < options.AUTO_HEAL_PERCENT){
-            auto_heal_from_menu_or_overworld(env.program_info(), env.console, context, 0, true);
-            stats.m_autoheals++;
-            env.update_stats();
-            send_program_status_notification(env, options.NOTIFICATION_STATUS_UPDATE);
-        }
-
-        /*
-        - Starts from pokemon center.
-        - Flies to start position. Runs a Let's Go iteration. 
-        - Then returns to pokemon center 
-        */
-        env.console.log("Starting Let's Go hunting path", COLOR_PURPLE);
-        int num_forward_moves_per_lets_go_iteration = options.NUM_FORWARD_MOVES_PER_LETS_GO_ITERATION;
-        run_from_battles_and_back_to_pokecenter(env, context, 
-            [&](
-                SingleSwitchProgramEnvironment& env, BotBaseContext& context
-            ){
-                run_until(
-                    env.console, context,
-                    [&](BotBaseContext& context){
-
-                        /*                         
-                        - run_from_battles_and_back_to_pokecenter will keep looping `action` 
-                        (i.e. this lambda function) until it succeeeds
-                        - Do a sandwich time check here to break out of the loop, in the case where
-                        you are very unlucky and can't finish a Let's Go iteration due to getting caught
-                        up in battles.
-                        */
-                        if (is_sandwich_expired(last_sandwich_time, minutes_per_sandwich)){
-                            env.log("Sandwich expired. Return to Pokecenter.");
-                            return;
-                        }                        
-                        move_to_start_position_for_letsgo1(env, context);
-                        run_lets_go_iteration(env, context, encounter_tracker, num_forward_moves_per_lets_go_iteration);
-                    },
-                    {hp_watcher}
-                );
-            } 
-        ); 
-        
-        context.wait_for_all_requests();
-    }
-
-
+    
+    return last_sandwich_time;
 }
 
-
-
-bool is_sandwich_expired(WallClock last_sandwich_time, std::chrono::minutes minutes_per_sandwich){
-    return (last_sandwich_time + minutes_per_sandwich) < current_time();
+// given the start time, and duration in minutes, check if the current time is
+// past the expected end time.
+bool is_time_expired(WallClock start_time, std::chrono::minutes minutes_duration){
```

Comment:
Did you intend to use this for checking program and sandwich expiration?

Reply:
It was used in an older version of the program. So, I can probably delete it.

Or maybe re-purpose it to calculate time_remaining, which could be used by farming_time_left and sandwich_time_left.